osafw-app/App_Code/models/S3.cs

```diff
@@ -109,15 +109,25 @@ public PutObjectResponse createFolder(string foldername)
     /// <param name="file">file from http upload</param>
     /// <param name="disposition">if defined (ex: inline) - Content-Disposition with file.FileName added</param>
     /// <param name="filename">optional filename to include in disposition header</param>
+    /// <param name="storage_class">S3 Storage Class, default is STANDARD, can be GLACIER_IR for warm archive files</param>
     /// <returns></returns>
-    public PutObjectResponse uploadLocalFile(string key, string filepath, string disposition = "", string filename = "")
+    public PutObjectResponse uploadLocalFile(string key, string filepath, string disposition = "", string filename = "", string storage_class = "STANDARD")
```

Comment:
How about just pass parameter as `S3StorageClass s3_storage_class=null` and then add to request if `!=null`
If your application uses AWS S3, you will have access to all S3 SDK classes so you will be able to pass `S3StorageClass.GlacierInstantRetrieval` or any other class.

And non-Standard class can be already configured for the bucket, if we pass null (or just nothing) we won't specify a class for the bucket, so the file will be stored under default's bucket class

Reply:
code updated